src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -137,3 +153,96 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
         if comment:
             query += f" comment='{comment}'"
         return self._execute_query(query)
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        parameters: Optional[List[str]] = None,
+    ):
+        all_files_list = self._get_files_list_from_stage(stage_path)
+        # filter files from stage if match stage_path pattern
+        filtered_file_list = self._filter_files_list(stage_path, all_files_list)
+        # sort filtered files in alphabetical order with directories at the end
+        sorted_file_list = sorted(
+            filtered_file_list, key=cmp_to_key(self._stage_files_comparator)
+        )
+
+        sql_parameters = self._parse_execute_parameters(parameters)
+        results = []
+        for file in sorted_file_list:
+            results.append(self._call_execute_immediate(file, sql_parameters, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = self.get_stage_name_from_path(stage_path)
+        files_list_result = self.list_files(stage_name).fetchall()
+
+        if len(files_list_result) == 0:
+            raise ClickException(f"No files found on stage '{stage_name}'")
+
+        return [f["name"] for f in files_list_result]
+
+    def _filter_files_list(
+        self, stage_path: str, files_on_stage: List[str]
+    ) -> List[str]:
+        stage_path = self.remove_stage_prefix(stage_path)
+
+        if not stage_path.__contains__("/"):
+            filtered_list = files_on_stage
+        elif glob.has_magic(stage_path):
+            filtered_list = fnmatch.filter(files_on_stage, stage_path)
+        elif files_on_stage.__contains__(stage_path):
+            filtered_list = [stage_path]
+        else:
+            filtered_list = fnmatch.filter(files_on_stage, f"{stage_path}*")
+
+        if not filtered_list:
+            raise ClickException(f"No files matched pattern '{stage_path}'")
+
+        return filtered_list
+
+    def _stage_files_comparator(self, file_path_1, file_path_2):
+        file_path_1_directories = file_path_1.count("/")
+        file_path_2_directories = file_path_2.count("/")
+
+        if file_path_1_directories > file_path_2_directories:
+            return 1
+        elif file_path_1_directories < file_path_2_directories:
+            return -1
+        elif file_path_1 >= file_path_2:
+            return 1
+        else:
+            return -1
+
+    def _parse_execute_parameters(
```

Comment:
`_parse_execute_parameters`, `_stage_files_comparator` and `_filter_files_list` don't use `self` properties - I'd define them as `@staticmethod` or even define outside `StageManager`

Reply:
Now only `_parse_execute_parameters` doesn't use `self`, I added `@staticmethod`